lib/PlaidClient.js

```diff
@@ -210,6 +211,44 @@ Client.prototype.getTransactions =
     }, options, cb);
   };
 
+// getAllTransactions(String, Date, Date, Object?, Function)
+Client.prototype.getAllTransactions =
+  function(access_token, start_date, end_date, cb) {
+    var transactions = [];
+    var offset = 0, total_transactions = null;
+    var self = this;
+
+    var handleResponse = function(transactionsResponse) {
+      total_transactions = transactionsResponse.total_transactions;
+      transactions.push(...transactionsResponse.transactions);
+      offset += transactionsResponse.transactions.length;
+    };
+
+    Promise.coroutine(function*() {
+      while (true) {
+        if (transactions.length === total_transactions) {
+          break;
+        }
+
+        yield self.getTransactions(
+          access_token,
+          start_date,
+          end_date,
+          {
+            count: 500, // largest allowed value
+            offset: offset,
+          }
+        ).then(handleResponse).catch(function(err) {
+          throw err;
+        });
+      }
+    })().then(function() {
+      return cb(null, transactions);
```

Comment:
this seems reasonable to me... I agree with you that we should be using the `Promise` api internally...